### ISSUE TITLE
fix pickling support

### ISSUE THREAD

**Issue by RyanDeRose-TomTom** [phillipd94/pyclothoids#2]
First of all, I love this package! I am getting great geometry out of it.

It would be really nice for Clothoid to support pickling. It looks like the intent was there (and maybe it worked in the past) since it has `__getstate__` and `__setstate__` methods, but they aren't quite working. I even threw in a simple `__reduce__` method, but there are more parameters than pickle allows.

Maybe you could bypass pickle's tuple length limit by making `__init__` accept a tuple of the 6 parameters, instead of the 6 of them separately, and then unpack it. I may play around with this if I have enough time and desperation.

Anyway, here is simple code to verify that it doesn't work:

```
from cmath import pi
import pickle
from pyclothoids import Clothoid

params = 0, 0, 0, 1, 1, pi/2
clothoid = Clothoid.G1Hermite(*params)
with open('test.pkl', 'wb') as f:
    pickle.dump(clothoid, f)

with open('test.pkl', 'rb') as f:
    new_clothoid = pickle.load(f)

print(new_clothoid)
```

and the traceback:
```
Traceback (most recent call last):
  File "clothoid_pickle_bug.py", line 13, in <module>
    print(new_clothoid)
  File "lib\site-packages\pyclothoids\clothoid.py", line 53, in __str__
    return 'Clothoid: ' + ''.join(map(lambda m,n:m + ':' + str(getattr(self,n)) + ' ',
  File "lib\site-packages\pyclothoids\clothoid.py", line 53, in <lambda>
    return 'Clothoid: ' + ''.join(map(lambda m,n:m + ':' + str(getattr(self,n)) + ' ',
  File "lib\site-packages\pyclothoids\clothoid.py", line 49, in __getattr__
    return getattr(self._ClothoidCurve,name)() #mimic property getter syntax
AttributeError: 'NoneType' object has no attribute 'XStart'
```

**Comment by phillipd94** [comment]
I'm glad you've found the package useful, and I appreciate the feedback!

Looks like this issue is caused by one of the classic blunders.  The setstate function was assigning the thinly wrapped C++ clothoid object to a temporary variable, then initializing and assigning the return value of the initialization function to the _ClothoidCurve attribute of the Python Clothoid.  Unfortunately, the C++ method doesn't have a return value and only mutates the curve it is a member of.  As a result, any unpickled Clothoid is being initialized with a NoneType.  The [billion dollar mistake](https://en.wikipedia.org/wiki/Tony_Hoare) strikes again.

Fortunately the fix is quite simple, I'll add some pickling and unpickling tests to my local test suite to make sure this doesn't regress in the future, then get the fix onto pypi as quickly as possible.

**Comment by phillipd94** [comment]
Fixed in version 0.1.3 now on PyPI
